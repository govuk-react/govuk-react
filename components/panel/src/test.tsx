import * as React from 'react';
import { mount } from 'enzyme';

import { Panel } from '.';

describe('Panel', () => {
  it('should render a panel without a body body', () => {
    const wrapper = mount(<Panel title="Example" />);

    expect(wrapper.text().includes('Example')).toBe(true);
  });

  it('should render a panel body that is a string', () => {
    const wrapper = mount(<Panel title="Example">body</Panel>);

    expect(wrapper.text().includes('body')).toBe(true);
  });

  it('matches wrapper snapshot', () => {
    const wrapper = mount(<Panel title="Example">body</Panel>);

    expect(wrapper).toMatchSnapshot('wrapper mount');
  });
});
