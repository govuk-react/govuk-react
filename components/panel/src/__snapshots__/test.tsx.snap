// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  margin-bottom: 15px;
  padding: 25px;
  border: 5px solid transparent;
  text-align: center;
  color: #ffffff;
  background: #00703c;
}

.c1 {
  margin-top: 0;
  margin-bottom: 30px;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 32px;
  line-height: 1.09375;
}

.c1:last-child {
  margin-bottom: 0;
}

.c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 24px;
  line-height: 1.0416666666666667;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    padding: 35px;
  }
}

@media print {
  .c1 {
    font-size: 32px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 48px;
    line-height: 1.0416666666666667;
  }
}

@media print {
  .c2 {
    font-size: 24px;
    line-height: 1.05;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 36px;
    line-height: 1.1111111111111112;
  }
}

<Panel
  title="Example"
>
  <src__StyledPanel>
    <div
      className="c0"
    >
      <src__StyledTitle>
        <h1
          className="c1"
        >
          Example
        </h1>
      </src__StyledTitle>
      <src__StyledBody>
        <div
          className="c2"
        >
          body
        </div>
      </src__StyledBody>
    </div>
  </src__StyledPanel>
</Panel>
`;
